crates/core/tedge_api/src/health.rs

```diff
@@ -92,14 +96,195 @@ impl ServiceHealthTopic {
     }
 }
 
+#[derive(Deserialize, Serialize, Debug, Default)]
+pub struct HealthStatus {
+    #[serde(default)]
+    pub status: Status,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum Status {
+    Up,
+    Down,
+    #[serde(untagged)]
+    Other(String),
+}
+
+impl Default for Status {
+    fn default() -> Self {
+        Status::Other("unknown".to_string())
+    }
+}
+
+impl Display for Status {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let status = match self {
+            Status::Up => "up",
+            Status::Down => "down",
+            Status::Other(val) if val.is_empty() => "unknown",
+            Status::Other(val) => val,
+        };
+        write!(f, "{}", status)
+    }
+}
+
 #[derive(Debug)]
 pub struct HealthTopicError;
 
+impl HealthStatus {
+    pub fn try_from_health_status_message(
+        message: &MqttMessage,
+        mqtt_schema: &MqttSchema,
+        bridge_service_topic: &Topic,
+    ) -> Result<Self, HealthTopicError> {
+        if let Ok((topic_id, Channel::Health)) = mqtt_schema.entity_channel_of(&message.topic) {
```

Comment:
Unrelated to this PR, but this highlights an issue I got several time: this method uses a specific `mqtt_schema` only to parse a topic. The MQTT root prefix doesn't really matter and it would be more convenient to have a `parse` method returning all the component of a topic is set (i.e. not only the topic_id and channel but also the root prefix).